parsing/location.ml

```diff
@@ -417,25 +561,55 @@ type report_printer = {
     Format.formatter -> (Format.formatter -> unit) -> unit;
 }
 
+let is_dummy_loc loc =
+  (* Fixme: this should be just [loc.loc_ghost] and the function should be
+     inlined below. However, currently, the compiler emits in some places ghost
+     locations with valid ranges that should still be printed. These locations
+     should be made non-ghost -- in the meantime we just check if the ranges are
+     valid. *)
+  loc.loc_start.pos_cnum = -1 || loc.loc_end.pos_cnum = -1
+
+(* It only makes sense to highlight (i.e. quote or underline the corresponding
+   source code) locations that originate from the current input. *)
+let is_quotable_loc loc =
+  not (is_dummy_loc loc)
+  && loc.loc_start.pos_fname = !input_name
+  && loc.loc_end.pos_fname = !input_name
```

Comment:
What are the sources of non-quotable locations? Lexer line directives? Inlining in the compiler? Errors spanning both a `.ml` and a `.mli` file?

Reply:
I think the two only cases are either:
- dummy locs / ghost locs leaking out of the compiler or a buggy ppx
- if you feed a `.ml` file to the compiler which contains lexer line directives indicating an other file. I think this is relatively rare -- in particular this is not what happens when using -pp or -ppx or a ppx driver. In that case, we do not try to quote the source code, since we do not know where that file might be, so I think it's better to not try and read random files on the filesystem.

Reply:
It could be nice to indicate these two scenarios in a comment, to store our understanding of what can happen for posterity.

Reply:
Done.